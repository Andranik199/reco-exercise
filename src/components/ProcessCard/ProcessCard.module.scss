.processCard {
  display: flex;
  flex-direction: column;
  background: #FFFFFF;
  border: 1px solid rgba(0, 0, 0, 0.1);
  box-shadow: 3.80976px 0.952441px 15px rgba(154, 160, 170, 0.15);
  max-width: 300px;
  max-height: 170px;
  padding: 15px;
  position: relative;
  width: 100%;

  h1 {
    font-size: 18px;
    line-height: 16px;
  }

  p {
    margin-top: 34px;
    font-weight: 400;
    font-size: 12px;
    line-height: 21px;
    color: #666666;
  }
}

.iconBlock {
  background-color: #B8D3EC;
  position: absolute;
  right: 0;
  top: 0;
  border-bottom-left-radius: 270px 151px;
  img {
    margin: 10px 20px;
  }
}
