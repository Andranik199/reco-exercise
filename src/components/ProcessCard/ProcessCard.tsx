import React from 'react';
import styles from './ProcessCard.module.scss';
import {MdsItem} from "../../store/slices/processes/processes.slice";

type ProcessCardProps = {
    item: MdsItem
}

export const ProcessCard: React.FC<ProcessCardProps> = ({item}) => {
    return (
        <div className={styles.processCard}>
            <div className={styles.iconBlock}>
                <img src={item.iconUrl} alt={item.iconUrl} />
            </div>
            <h1>
                {item.name}
            </h1>

            <p>
                {item.description}
            </p>
        </div>
    )
};
