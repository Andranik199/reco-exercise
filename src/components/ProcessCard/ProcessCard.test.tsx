import {ProcessCard} from './ProcessCard'
import financeIcon from 'icons/finance.svg'
import {create} from "react-test-renderer";

it("Should render ProcessCard component properly", () => {
    const component = create(<ProcessCard item={{
        id: '1',
        name: 'Finance',
        description: 'This process exmains the payroll flow within the Finanace department. This process was created for Ziv Cohen on 06.07.2021, 17:58',
        iconUrl: financeIcon
    }}/>)
    expect(component.toJSON()).toMatchSnapshot()
})
