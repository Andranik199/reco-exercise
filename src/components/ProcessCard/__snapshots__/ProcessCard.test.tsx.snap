// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ProcessCard component properly 1`] = `
<div
  className="processCard"
>
  <div
    className="iconBlock"
  >
    <img
      alt="finance.svg"
      src="finance.svg"
    />
  </div>
  <h1>
    Finance
  </h1>
  <p>
    This process exmains the payroll flow within the Finanace department. This process was created for Ziv Cohen on 06.07.2021, 17:58
  </p>
</div>
`;
